### PR TITLE
Refactored to use registry

### PR DESCRIPTION
**Pull request by mvpoyatt**
## Release Notes

- **New Features**
  - Enhanced `ChannelDetails` and `PacketDetails` components to dynamically fetch and display blockchain explorer URLs based on the selected chains.
  - Introduced a new API endpoint for retrieving blockchain explorer URLs based on specified parameters.
  - Added functionality to fetch chain IDs and explorer URLs from a centralized registry.
  - Implemented a new utility function to determine the client environment for fetching explorer URLs.

- **Bug Fixes**
  - Improved error handling for fetching fee data and explorer URLs, ensuring more informative responses for invalid requests.

- **Refactor**
  - Streamlined state management in components by replacing outdated state variables with more descriptive names.
  - Centralized registry fetching logic to improve maintainability.
  - Removed unnecessary properties from chain configuration and type definitions to simplify structure.

- **Documentation**
  - Updated internal documentation to reflect new functions and API changes.

### CODE REVIEW

File: app/api/registry/fetch-registry.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    if (process.env.GITHUB_TOKEN) {
+      res = await fetch(process.env.REGISTRY_URL!, {
+        headers: {
+          Authorization: `token ${process.env.GITHUB_TOKEN}`,
+        },
+        cache: 'no-store'
+      });
+    } else {
+      res = await fetch(process.env.REGISTRY_URL!, { cache: 'no-store' });
+    }
```
[the remark this answers is not in the document]
Comment (dshiell) [reply app/api/registry/fetch-registry.ts:18]: @mvpoyatt wdyt of ai suggestion?

File: app/api/registry/fetch-registry.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    data = await res.json();
+  }
+  catch (error) {
+    logger.error('Error fetching polymer-registry: ' + error);
+    return {};
+  }
+
+  return data;
+}
```
[the remark this answers is not in the document]
Comment (dshiell) [reply app/api/registry/fetch-registry.ts:32]: worth doing?

File: app/api/registry/route.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  let env = searchParams.get('env') || '';
+
+  if (!chainName || !info || !env) {
+    return NextResponse.json({ error: 'Invalid request' }, { status: 400 });
+  }
+
+  let chainId: number = getChainId(chainName, env);
+  if (!chainId) {
+    return NextResponse.json({ error: 'Invalid environment or chain name' }, { status: 400 });
+  }
```
[the remark this answers is not in the document]
Comment (dshiell) [reply app/api/registry/route.ts:22]: I don't think chainId can be '0', but technically valid concern.

**Review by dshiell — APPROVED** [review]
lgtm, maybe take a look through some of the coderabbit suggestions, but we can start testing in staging once ready!

File: app/api/registry/fetch-registry.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mvpoyatt) [reply app/api/registry/fetch-registry.ts:18]: @dshiell probably a good idea, I'll make this change

**Pull request merged**